File: bidragsevne/src/main/java/no/nav/bidrag/beregn/bidragsevne/BidragsevneCore.java
package no.nav.bidrag.beregn.bidragsevne;


import no.nav.bidrag.beregn.bidragsevne.dto.BeregnBidragsevneGrunnlagAltCore;
import no.nav.bidrag.beregn.bidragsevne.dto.BeregnBidragsevneResultatCore;

public interface BidragsevneCore {

  BeregnBidragsevneResultatCore beregnBidragsevne(BeregnBidragsevneGrunnlagAltCore grunnlag);

  static BidragsevneCore getInstance() {
    return new BidragsevneCoreImpl();
  }
}

File: bidrag-beregn-særtilskudd/src/main/kotlin/no/nav/bidrag/beregn/bidragsevne/BidragsevneCore.kt
package no.nav.bidrag.beregn.bidragsevne

import no.nav.bidrag.beregn.bidragsevne.dto.BeregnBidragsevneGrunnlagCore
import no.nav.bidrag.beregn.bidragsevne.dto.BeregnBidragsevneResultatCore
import no.nav.bidrag.beregn.bidragsevne.periode.BidragsevnePeriode

fun interface BidragsevneCore {
    fun beregnBidragsevne(grunnlag: BeregnBidragsevneGrunnlagCore): BeregnBidragsevneResultatCore

    companion object {
        fun getInstance(): BidragsevneCore = BidragsevneCoreImpl(BidragsevnePeriode.getInstance())
    }
}
